DotNetExcel/cardPurchase.cs

```diff
@@ -0,0 +1,219 @@
+            worksheet.Range["H" + TableFormat.HeaderRow].Value = "Anexo";
+            worksheet.Range["I" + TableFormat.HeaderRow].Value = "ID Cartão";
+            worksheet.Range["J" + TableFormat.HeaderRow].Value = "ID Holder";
+
+            Dictionary<string, object> optionalParam = new Dictionary<string, object>();
+
+            int row = TableFormat.HeaderRow + 1;
+
+            string cursor = "";
+            Dictionary<string, object> returnedData = new Dictionary<string, object>();
+
+            do
+            {
+
+                JObject respJson;
+
+                try
+                {
+                    respJson = corporatePurchase.Get(cursor);
+                }
+                catch (Exception ex)
+                {
+                    MessageBox.Show(ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                    return;
+                }
+
+                if ((string)respJson["cursor"] != "") cursor = (string)respJson["cursor"];
+
+                JArray purchases = (JArray)respJson["purchases"];
+
+                foreach (JObject purchase in purchases)
+                {
+
+                    worksheet.Range["A" + row].Value = purchase["created"];
+                    worksheet.Range["B" + row].Value = purchase["id"];
+                    worksheet.Range["C" + row].Value = purchase["merchantCategoryCode"];
+                    worksheet.Range["D" + row].Value = purchase["merchantName"];
+                    worksheet.Range["E" + row].Value = purchase["description"];
+                    worksheet.Range["F" + row].Value = purchase["status"];
+                    worksheet.Range["G" + row].Value = purchase["amount"];
+
+                    try
+                    {
+                        foreach (JToken attachment in purchase["attachments"])
+                        {
+                            worksheet.Range["H" + row].Value = attachment["id"] + "," + worksheet.Range["H" + row].Value;
+                        }
+                        worksheet.Range["H" + row].Value = worksheet.Range["H" + row].Value.Substring(0, worksheet.Range["H" + row].Value.Length - 1);
+                    }
+                    catch { }
+
+                    worksheet.Range["I" + row].Value = purchase["cardId"];
+                    worksheet.Range["J" + row].Value = purchase["holderId"];
+                    worksheet.Range["K" + row].Value = purchase["cenderId"];
+
+                    row++;
+                }
+
+            } while (cursor != null);
+        }
+
+        private void button6_Click(object sender, EventArgs e)
+        {
+            var worksheet = Globals.Planilha12;
+            string selectedPath = "";
+
+            Excel.Range selectedRange = worksheet.Application.Selection;
+
+            if (selectedRange.Address.Substring(1, 1) == "H")
+            {
+                JObject respJson;
+                int range_a = int.Parse(selectedRange.Address.Substring(3).Split(':')[0]);
```

Comment:
Nomes de variáveis não estão explicativos range_a e range_b, por que ele é do tipo int mas o nome está indicando como range
Coloque algo mais indicativo para demonstrar a ideia de que ele será utilizado no For.